ARCHITECTURE.md

```diff
@@ -0,0 +1,97 @@
+
+Kubernetes is a datacenter-scale operating system.  The configuration files for kubernetes allow for source-controlled management of datacenter-level functionality.  This includes path-based routing of http requests to the various back-end applications; authentication of the same requests to a 3rd-party identity management service; monitoring of performance and health metrics, alerting on failure conditions; and demand-based scaling of compute resources.
+# Deployment
+We chose [kubernetes](https://kubernetes.io/) as a widely-supported, vendor-agnostic method of deploying containers into a cloud. A custom tool, mdsctl, provides a wrapper around [kubectl](https://github.com/kubernetes/kubectl) and (helm)[https://helm.sh/] enabling portable build, deployment, development and verification operations.
+# Database & ORM
+Persistent storage is contained in Postgres. The current interface for each service is performed via a hand-made ORM in the mds-db library, but we are in the process of replacing it with a more robust and FOSS library, TypeORM. See the documentation in the mds-db package for more details, although this will be phased out in the next release.
+
+DB credentials are injected to services via environment variables during the deployment cycle.
+# Cache
+For frequently-accessed data, such as the current state of all active vehicles in the system, the mds-cache library uses Redis under the hood.  
+
+The interface to the cache is described in the package mds-cache.  
+# Messaging
+Although the current services in MDS-core do not communicate with each other, LADOT is using some data-consumers that process Agency-ingested events and telemetry via a publish-and-subscribe message bus.
+
+Services in a service-oriented-architecture can communicate with each other via the same types of interfaces used by the outside clients (e.g. REST or SOAP), or they can use an internal message bus. We chose the latter for perceived wins in speed and efficiency. After considering knative eventing and kafka, we settled on [NATS](https://nats.io) as a more stable and low-operational-overhead solution. We have attempted to abstract away the message interface, however, such that we could later decide to plug in a different message bus.
+
+NATS is recognized as an incubating [CNCF](https://www.cncf.io) [open source project](https://github.com/nats-io) made available under the [Apache License](https://www.apache.org/licenses/LICENSE-2.0.html) and supporting a number of contemporary software languages.
+# Testing & Validation
+The repo currently has integration tests with code-coverage minimums on a per-project basis.  [TODO add more]
+# Versioning
+The services in mds-core will follow the MDS spec’s requirements for [API versioning](https://github.com/openmobilityfoundation/mobility-data-specification/blob/dev/general-information.md).
+# Shared Libraries
```

Comment:
This will need to be updated to add `mds-service-helpers` and `mds-webpack-config`